go/vt/callerid/callerid.go

```diff
@@ -26,14 +26,15 @@ import (
 )
 
 // The datatype for CallerID Context Keys
-type callerIDKey int
-
-var (
-	// internal Context key for immediate CallerID
-	immediateCallerIDKey callerIDKey
-	// internal Context key for effective CallerID
-	effectiveCallerIDKey callerIDKey = 1
-)
+type callerIDContextKey struct{}
+
+// callerIDContext is a tuple containing the entirety
+// of the "callerid" such that both pointers can be
+// stored together.
+type callerIDContext struct {
+	ef *vtrpcpb.CallerID
+	im *querypb.VTGateCallerID
```

Comment:
I'd call these `effective` and `immediate` for (a) readability and (b) so that people can correlate them to concepts they are already familiar with.

Reply:
Yeah, I can do that. I yanked `ef` and `im` since they were used throughout this file already for function args, etc.